tests/unit/CircuitBreaker.spec.ts

```diff
@@ -0,0 +1,283 @@
+import { afterEach, beforeEach, describe, expect, test, jest } from '@jest/globals';
+
+import { CircuitBreaker, BreakerState } from '../../src/CircuitBreaker';
+import { defaultProbeEvaluateInterval } from '../../src/Configuration';
+
+describe('CircuitBreaker', () => {
+    let b: CircuitBreaker;
+
+    beforeEach(() => {
+        jest.useFakeTimers();
+    });
+
+    afterEach(() => {
+        if (b?._evaluateTimeoutHandle) {
+            clearTimeout(b._evaluateTimeoutHandle);
+        }
+    });
+
+    describe('default configuration', () => {
+        test('should init', () => {
+            expect(() => new CircuitBreaker({})).not.toThrow();
+        });
+
+        test('should always be in nominal state', () => {
+            b = new CircuitBreaker({});
+            b._evaluate();
+            expect(b.state).toStrictEqual(BreakerState.Nominal);
+        });
```

Comment:
I'm not sure I get the question; breakers always start in `nominal` case, so in case of a restart, some activity is allowed to take place, so no "wait". This is for protection against probes failing to gather data when the process starts; it's best for the breaker to become transparent (if prometheus were down, ops would be alerted anyways) than to block all activity waiting for a fix.

Reply:
This is aligned with the design, I get the point here, thank you.
My point was mostly about whether we want to avoid at all cost an overloaded platform (that may be the case between the circuit breaker starts and the time it takes to change its state), or if we want to be more flexible as you described.